brave/brave-core
# [iOS] Edit Custom Search Engines

Resolves https://github.com/brave/brave-browser/issues/39552

This PR is adding the functionality of edit to custom search screen and engines.

In addition to delete now edit is added to the swipe functionality. The same screen is used including the same error checks for this action except the duplicate error check since the engine is edited. In addition the view controller functionality is divided into chunks to make edit easier for next iteration.



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- Add custom search engine
- Use the template search engine mentioned in the description
- Try edit functionality and change url or title 
- Check if the changes are done
- Test query with them

## Screenshots:

https://github.com/user-attachments/assets/1bf4ea15-c207-4409-8bbb-ade3f1296c07
